Add progress circle to index list view

ReeceHoffmann opened the pull request:
Changes:

 - Adds progress indicators to Index list view
 - Updates list view to use react-query

ReeceHoffmann committed:
feat: add index creation progress indicators in index list view

ReeceHoffmann committed:
Merge branch 'main' into add-progress-to-indexes

ReeceHoffmann committed:
chore: remove redundant unit tests

ReeceHoffmann committed:
chore: ensure tsc compliance

igboyes commented on src/js/app/websocket.ts:
```diff
@@ -17,21 +17,60 @@ import { accountKeys } from "../account/querys";
     roles: roleQueryKeys,
     uploads: fileQueryKeys,
     users: userQueryKeys,
+    indexes: indexQueryKeys,
```
Maintain alpha sort.

igboyes commented on src/tests/fake/indexes.ts:
```diff
@@ -0,0 +1,78 @@
+};
+
+export function createFakeIndexNested(props?: CreateFakeIndexNestedProps): IndexNested {
+    const defaultIndexNested = {
+        id: faker.random.alphaNumeric(8),
+        version: faker.datatype.number(5),
+    };
+
+    return merge(defaultIndexNested, props);
```
Same thing?
```suggestion
    return {
        id: faker.random.alphaNumeric(8),
        version: faker.datatype.number(5),
        ...props
    };
```

ReeceHoffmann replied:
Yeah it basically does, but if props is undefined it throws an error. This could be easily fixed by assigning an empty object to props if nothing is passed. But this soln reads more clearly to me

igboyes commented on src/tests/fake/indexes.ts:
```diff
@@ -0,0 +1,78 @@
+    return merge(defaultIndexNested, props);
+}
+
+type createFakeIndexMinimalProps = CreateFakeIndexNestedProps & {
```
I believe this is TS convention. This inconsistency is introduced in a few other spots as well.
```suggestion
type CreateFakeIndexMinimalProps = CreateFakeIndexNestedProps & {
```

igboyes commented on src/js/indexes/components/RebuildAlert.tsx:
```diff
@@ -3,9 +3,17 @@ import { connect } from "react-redux";
 
-export const RebuildAlert = ({ refId, showCountAlert, showIndexAlert }) => {
-    if (showCountAlert === 0) {
+export function RebuildAlert({ refId, hasRights }) {
```
Shouldn't typing go here?

igboyes commented on src/js/app/websocket.ts:
```diff
@@ -17,21 +17,60 @@ import { accountKeys } from "../account/querys";
 import { groupQueryKeys } from "../groups/querys";
+import { indexQueryKeys } from "../indexes/querys";
 
+/** Get effected resource query keys from job name  */
```
```suggestion
/** Get affected resource query keys by workflow name.  */
```

igboyes commented on src/js/app/websocket.ts:
```diff
@@ -17,21 +17,60 @@ import { accountKeys } from "../account/querys";
+};
+
+/**
+ * Invalidate queries that are effected by a job update
```
```suggestion
 * Invalidate queries that are affected by a job update
```

igboyes commented on src/tests/fake/references.ts:
```diff
@@ -0,0 +1,19 @@
+import { merge } from "lodash";
+import { ReferenceDataType } from "../../js/references/types";
+
+type createFakeReferenceNestedProps = {
```
```suggestion
type CreateFakeReferenceNestedProps = {
```

igboyes commented on src/tests/fake/utils.ts:
```diff
@@ -0,0 +1,5 @@
+export type FakeSearchResults = {
+    page_count?: number;
+    per_page?: number;
+    total_count?: number;
```
Shouldn't these properties always be defined?

ReeceHoffmann replied:
Ah no, which points to a naming mistake. The idea here was that it would work as a base model for allowing the fake functions to take optional manual overrides for the search result data returned by nock. `fakeSearchResultOptions`?

igboyes reviewed (changes requested):
* Use `function` in touched modules.
* Follow one component export per module. This optimizes the Vite build and HMR.

ReeceHoffmann committed:
chore: address review

- Reduce usage of arrow notation
- fix docstrings and add
- Ensure one component per file in changed files
- Add typing where missing

ghost commented:
:tada: This PR is included in version 3.16.0 :tada:

The release is available on [GitHub release](https://github.com/virtool/virtool-ui/releases/tag/3.16.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
